realm/src/androidTest/java/io/realm/RealmQueryTest.java

```diff
@@ -459,4 +461,43 @@ public void testGeorgian() {
             assertEquals(sorted[i], stringOnlies2.get(i).getChars());
         }
     }
+
+    private RealmResults<CatOwner> results = null;
+    public void testModifyRealmListD() {
+        // If it doesn't crash, increase this
+        final int count = 100;
+        // Enable the list for maintaining the reference will avoid crash!!!!!!!!!!!!!
+        //final ArrayList<RealmResults<CatOwner>> arrayList = new ArrayList<RealmResults<CatOwner>>();
+
+        results = testRealm.where(CatOwner.class).findAll();
+        RealmChangeListener listener = new RealmChangeListener() {
+            @Override
+            public void onChange() {
+                // Enable the list for maintaining the reference will avoid crash!!!!!!!!!!!!!
+                // arrayList.add(results);
+                results = results.where().notEqualTo("name", "asdfas" + count).findAll();
+                //assertEquals(count, results.size());
+                for (int i=0; i<count; i++) {
+                    CatOwner catOwner = results.get(i);
+                    assertEquals(catOwner.getName(), "catOwner"+i);
+                    Log.e("testModifyRealmListD", "loop count " + i);
+                }
+                System.gc();
```

Comment:
This is just a hint to the GC. You are not guaranteed that it runs. So adding a note about what this is trying to trigger would be nice.